lib/AboutPreferences.jsm

```diff
@@ -269,77 +267,9 @@ this.AboutPreferences = class AboutPreferences {
         subcheck.classList.add("indent");
         subcheck.setAttribute("label", formatString(nested.titleString));
         linkPref(subcheck, nested.name, "bool");
-        if (nested.name === PREF_SHOW_SPONSORED) {
-          sponsoredStoriesCheckbox = subcheck;
-        }
       });
     });
 
-    if (discoveryStreamConfig.enabled) {
-      // If Discovery Stream is enabled hide Home Content options
-      contentsGroup.style.visibility = "collapse";
-
-      const discoveryGroup = homeGroup.insertAdjacentElement("afterend", homeGroup.cloneNode());
-      discoveryGroup.id = "discoveryContentsGroup";
-      discoveryGroup.setAttribute("data-subcategory", "discovery");
-      createAppend("label", discoveryGroup)
-        .appendChild(document.createElementNS(HTML_NS, "h2"))
-        .textContent = formatString("prefs_content_discovery_header");
-      const descriptionHbox = createAppend("hbox", discoveryGroup);
-      const discoveryGroupDescription = createAppend("description", descriptionHbox);
-      discoveryGroupDescription.textContent = formatString("prefs_content_discovery_description");
-      discoveryGroupDescription.classList.add("tail-with-learn-more");
-
-      // Add the Learn more link in the description
-      const topstoriesSection = prefStructure.find(s => s.id === "topstories");
-      const learnMoreURL = topstoriesSection && topstoriesSection.learnMore.link.href;
-      const link = createAppend("label", descriptionHbox);
-      link.classList.add("learn-sponsored");
-      link.classList.add("text-link");
-      link.setAttribute("href", learnMoreURL);
-      link.textContent = formatString("prefs_topstories_sponsored_learn_more");
-
-      if (discoveryStreamConfig.show_spocs && sponsoredStoriesCheckbox) {
```

Comment:
DiscoveryStreamFeed shows and hide spocs based off pref showSponsored.
https://searchfox.org/mozilla-central/source/browser/components/newtab/lib/DiscoveryStreamFeed.jsm#74

Do we need to explicitly check in about:preferences going forward for discoveryStreamConfig.show_spocs if discovery stream is enabled or handling via showSponsored pref suffice?

Reply:
Just to document, the nested sponsored prefs only show up when spocs are enabled:
https://github.com/mozilla/activity-stream/blob/8b20ec0902e5109303c2fe30a32fda7cf33d74e8/lib/SectionsManager.jsm#L24-L28